core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -121,6 +131,54 @@ void testAbsoluteFileToUrl(String path, String expectedUrl) throws URISyntaxExce
         () -> "Sources.of(Sources.of(file(" + path + ").absolutePath).url()).file().getPath()");
   }
 
+  /**
+   * Resources created with the JAR protocol should also work.
+   * This will enable the Fixture test to work under Bazel.
+   */
+  @Test void testJarFileUrl() throws  MalformedURLException {
+    // mock jar file
+    final String jarPath = "jar:file:sources!/abcdef.txt";
+    final URL url = new URL(jarPath);
+    final Source source = of(url);
+    assertThat("No file retrieved for Sources.of(file " + jarPath + ")",
+        source.file(), notNullValue());
+    assertThat("Sources.of(file " + jarPath + ").url()).file().getPath()",
+        slashify(source.file().getPath()),
+        is("sources!/abcdef.txt"));
+
+  }
+
+  /** Tests {@link Sources#of(URL)} with code similar to
+   * {@code DiffRepository.Key#toRepo()}. */
+  @Test void testJarFileUrlWrite() {
+    final URL refFile = FoodmartQuerySet.class.getResource("/queries.json");
```

Comment:
To be honest I'm a bit confused at what all is going on here. Would you mind either:
- Extracting some of the setup into a helper (maybe easier testing is possible since this PR was opened).
- Breaking this into multiple tests
- Adding more comments

If it's all common sense and I'm OOTL then disregard.

Reply:
Crap it is so long ago I don't remember why I did that to test the diff repo